CMakeLists.txt

```diff
@@ -176,15 +176,15 @@ elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "
   target_compile_options(azure-storage-lite PRIVATE -Wall -Wextra -Werror -pedantic)
 endif()
 
-if(BUILD_ADLS)
+if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME AND BUILD_ADLS)
```

Comment:
maybe we don't need to check project name for this option?

Reply:
makes sense, my fault.